core/lib/tasks/exchanges.rake

```diff
@@ -20,12 +20,13 @@ namespace :exchanges do
       begin
         item_charger.charge_for_items
       rescue Spree::UnreturnedItemCharger::ChargeFailure => e
-        failure = {message: e.message, new_order: e.new_order.try(:number)}
-      rescue Exception => e
-        failure = {message: "#{e.class}: #{e.message}"}
+        failure = { message: e.message }
+      rescue => e
```

Comment:
@magnusvk @gmacdougall changing this to `rescue => e` instead of `rescue Exception => e` does seem like the right move. (I think `rescue Exception` was a bad habit of mine from the Ruby 1.8 days where `Timeout::Error` (and maybe some other classes like that?) did not inherit from StandardError and it was often bad that `rescue => e` did not rescue them.)

But with this update we now have the problem that our `failure_handler` will never be called for anything in `failures` if a non-StandardError occurs while `failures` has something in it. (e.g. if you sigint the rake task to stop it.

We could add:
```ruby
rescue Exception
  Spree::UnreturnedItemCharger.failure_handler.call(failures) if Spree::UnreturnedItemCharger.failure_handler
  raise
end
```
but that might not be good, depending on what the non-StandardError is.

Or we could add more rescues like `rescue Interrupt => e ...` for expected scenarios.
Or we could add signal handling stuff to this rake task specifically so that you can kill the task and run cleanup code.

Or we could change our failure handler so that it was called right away for each error as it occurred.  That seems like the most sane option to me.

Or we could just keep this as it is now and not worry about any of this.

Thoughts?